app/index.js

```diff
@@ -44,7 +44,10 @@ module.exports = generators.Base.extend({
         type: 'input',
         name: 'app_id',
         message: 'App ID',
-        default: answers => answers.name
+        validate: (input) => {
+          return (/\-|\s/.test(input)) ? 'Invalid App ID, use alphanumeric and periods only' : true;
```

Comment:
I don't have a spec. The [cordova doc](https://cordova.apache.org/docs/en/latest/config_ref/index.html#widget) calls it the app's *"reverse-domain identifier"*. I've learned that, for Android, it also has to be a valid Java package name, i.e. it must not start with a number, and must not contain reserved Java keywords such as `int` as package segments.

I'd prefer the regex to be rather strict for a start. We can still adjust it on request. After all, it's a template and not everything that is permitted is also recommendable. However, covering those exceptions like the Java keywords seem to be out of scope. We don't want to maintain this list of keywords, do we? I'd suggest something simple like:

```js
/^([a-z0-9_]+\.)*[a-z][a-z0-9_]$/i
```

Does that make sense?